### PR TITLE
add func:  entr_detr_env_moisture_deficit_div

### PR DESCRIPTION
**Pull request by jiahe23**
in turbulence_functions.pyx:  entr_detr_env_moisture_deficit_div that adds 1/M dM/dz on top of the fractional entrainment/detrainment calculated by entr_detr_env_moisture_deficit

this function is called by setting the 'entrainment' namelist option as 'moisture_deficit_div'

### CODE REVIEW

File: Turbulence_PrognosticTKE.pyx [hunk 1]
```diff
@@ -1353,6 +1360,14 @@ cdef class EDMF_PrognosticTKE(ParameterizationBase):
                     input.RH_upd = self.UpdVar.RH.values[i,k]
                     input.RH_env = self.EnvVar.RH.values[k]
 
+                    # compute dMdz at half levels
+                    gmv_w_k = interp2pt(GMV.W.values[k], GMV.W.values[k-1])
```
Comment (yairchn) [inline Turbulence_PrognosticTKE.pyx:1364]: isn't GMV.W zero by definition ? you choose to have it here for completeness ?

File: Turbulence_PrognosticTKE.pyx [hunk 2]
```diff
@@ -149,6 +151,10 @@ cdef class EDMF_PrognosticTKE(ParameterizationBase):
         self.surface_area = paramlist['turbulence']['EDMF_PrognosticTKE']['surface_area']
         self.max_area = paramlist['turbulence']['EDMF_PrognosticTKE']['max_area']
         self.entrainment_factor = paramlist['turbulence']['EDMF_PrognosticTKE']['entrainment_factor']
+        try:
+            self.entrainment_Mdiv_factor = paramlist['turbulence']['EDMF_PrognosticTKE']['entrainment_Mdiv_factor']
```
Comment (yairchn) [inline Turbulence_PrognosticTKE.pyx:155]: I would make it more verbose, "entrainment_momentum_div_factor" ?

Comment (jiahe23) [reply Turbulence_PrognosticTKE.pyx]: I changed it to "entrainment_massflux_div_factor"

File: Turbulence_PrognosticTKE.pyx (hunk 1 above)
Comment (jiahe23) [reply Turbulence_PrognosticTKE.pyx:1364]: Yes, this won't affect the experiment results in SCM. But we need to discussed the correct formula when implementing to GCM

**Pull request merged**